src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -164,9 +164,6 @@ def run(self, inputs: List[numpy.ndarray], val_inp: bool) -> List[numpy.ndarray]
 
         if self.internal_cache_active:
             # validate the inputs if needed
-            if val_inp:
-                self.engine._validate_inputs(inputs)
```

Comment:
We need to remove this, otherwise, we get an error:
```bash
ValueError: array batch size of 0 must match the batch size the model was instantiated with 1
```